src/clj/game/core/cost_fns.clj

```diff
@@ -41,11 +41,12 @@
           (max 0)))))
 
 (defn rez-additional-cost-bonus
-  [state side card]
-  (merge-costs
-    (concat (:additional-cost card)
-            (:additional-cost (card-def card))
-            (get-effects state side :rez-additional-cost card))))
+  ([state side card] (rez-additional-cost-bonus state side card (constantly true)))
+  ([state side card pred]
+   (filter pred (merge-costs
```

Comment:
I'd prefer `(let [costs (merge-costs ...)] (if pred (filterv pred costs) costs))`. No need to do extra work.

Reply:
happy with either, kinda took the first one to be more functional / less having to declare extra state but also think the `if` does read more straightforward